pkg/splunk/enterprise/util.go

```diff
@@ -681,14 +728,15 @@ func isAppExtentionValid(receivedKey string) bool {
 }
 
 // AddOrUpdateAppSrcDeploymentInfoList  modifies the App deployment status as perceived from the remote object listing
-func AddOrUpdateAppSrcDeploymentInfoList(appSrcDeploymentInfo *enterprisev1.AppSrcDeployInfo, remoteS3ObjList []*splclient.RemoteObject) bool {
+func AddOrUpdateAppSrcDeploymentInfoList(appSrcDeploymentInfo *enterprisev1.AppSrcDeployInfo, remoteS3ObjList []*splclient.RemoteObject, appFrameworkConfig *enterprisev1.AppFrameworkSpec, appSrc string, cr splcommon.MetaObject, client splcommon.ControllerClient) bool {
```

Comment:
@sgontla I added these so that I could handle the download/copy/install inline in this function.  I know this isn't the right place for it so I many need so help in finding it a proper home.